src/celeritas/em/msc/detail/UrbanMscScatter.hh

```diff
@@ -397,15 +389,23 @@ CELER_FUNCTION real_type UrbanMscScatter::sample_cos_theta(Engine& rng) const
         return max(result, real_type(1.9));
     }();
 
-    real_type ea = std::exp(-xsi);
     // Mean of cos\theta computed from the distribution g_1(cos\theta)
     // small theta => x = theta0^2
-    // large xsi => xmean_1 = 1 - x
+    // large xsi => xmean_1 = 1
     // small tau => xmean = 1
     real_type x = ipow<2>(2 * std::sin(real_type(0.5) * theta0_));
-    real_type xmean_1 = 1 - x * (1 - (1 + xsi) * ea) / (1 - ea);
 
-    if (xmean_1 <= real_type(0.999) * xmean_)
+    // Calculate intermediate values for the mean of cos(theta)
+    // Since xsi is not near zero (thanks to max), no need to use expm1
+    // The expression in outer parens is in [~0.666, 1]
+    real_type xmean_1 = 1 - x * (1 - xsi / (std::exp(xsi) - 1));
```

Comment:
let `ex = exp(xsi) = 1/ea`
```
xmean_1
= 1 - x * (1 - (1 + xsi) * ea) / (1 - ea)  // multiply parens by ex/ex
= 1 - x * (ex - (1 + xsi)) / (ex - 1)
= 1 - x * (ex - 1 - xsi) / (ex - 1)
= 1 - x *( (ex - 1)/(ex - 1) - xsi / (ex - 1) )
= 1 - x *(1 - xsi / (ex - 1))
```

Reply:
I actually have no idea how it's implemented... but I would assume there's more optimizations for `std::exp` than the special case.